contracts/traderjoe/FarmLens.sol

```diff
@@ -0,0 +1,209 @@
+        uint256 priceFromWavaxDai = _getAvaxPrice(IJoePair(address(0xA389f9430876455C36478DeEa9769B7Ca4E3DDB1))); // 18
+
+        uint256 sumPrice = priceFromWavaxUsdt.add(priceFromWavaxUsdc).add(priceFromWavaxDai); // 18
+        uint256 avaxPrice = sumPrice / 3; // 18
+        return avaxPrice; // 18
+    }
+
+    function _getAvaxPrice(IJoePair pair) private view returns (uint256) {
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+
+        if (pair.token0() == wavax) {
+            reserve1 = reserve1.mul(_tokenDecimalsMultiplier(pair.token1())); // 18
+            return (reserve1.mul(1e18)) / reserve0; // 18
+        } else {
+            reserve0 = reserve0.mul(_tokenDecimalsMultiplier(pair.token0())); // 18
+            return (reserve0.mul(1e18)) / reserve1; // 18
+        }
+    }
+
+    function getPriceInUSD(address tokenAddress) public view returns (uint256) {
+        return (getAvaxPrice().mul(getPriceInAvax(tokenAddress))) / 1e18; // 18
+    }
+
+    /// @dev Need to be aware of decimals here, not always 18, it depends on the token
+    function getPriceInAvax(address tokenAddress) public view returns (uint256) {
+        if (tokenAddress == wavax) {
+            return 1e18;
+        }
+
+        IJoePair pair = IJoePair(joeFactory.getPair(tokenAddress, wavax));
+
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+        address token0Address = pair.token0();
+        address token1Address = pair.token1();
+
+        if (token0Address == wavax) {
+            reserve1 = reserve1.mul(_tokenDecimalsMultiplier(token1Address)); // 18
+            return (reserve0.mul(1e18)) / reserve1; // 18
+        } else {
+            reserve0 = reserve0.mul(_tokenDecimalsMultiplier(token0Address)); // 18
+            return (reserve1.mul(1e18)) / reserve0; // 18
+        }
+    }
+
+    function _tokenDecimalsMultiplier(address tokenAddress) private pure returns (uint256) {
+        uint256 decimalsNeeded = 18 - IJoeERC20(tokenAddress).decimals();
+        return 1 * (10**decimalsNeeded);
+    }
+
+    function getReserveUSD(IJoePair pair) public view returns (uint256) {
+        address token0Address = pair.token0();
+        address token1Address = pair.token1();
+
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+
+        reserve0 = reserve0.mul(_tokenDecimalsMultiplier(token0Address)); // 18
+        reserve1 = reserve1.mul(_tokenDecimalsMultiplier(token1Address)); // 18
+
+        uint256 token0PriceInAvax = getPriceInAvax(token0Address); // 18
+        uint256 token1PriceInAvax = getPriceInAvax(token1Address); // 18
+        uint256 reserve0Avax = reserve0.mul(token0PriceInAvax); // 36;
+        uint256 reserve1Avax = reserve1.mul(token1PriceInAvax); // 36;
+        uint256 reserveAVAX = (reserve0Avax.add(reserve1Avax)) / 1e18; // 18
+        uint256 reserveUSD = (reserveAVAX.mul(getAvaxPrice())) / 1e18; // 18
+
+        return reserveUSD; // 18
```

Comment:
Not a fan of using caps for `AVAX` and `USD`, when previous vars are named with `Avax`. 

Suggest change `reserveAVAX` to `totalReserveAvax` and `reserveUSD` to `totalReserveUsd`